apps/emqx_bridge_mqtt/src/emqx_bridge_mqtt_connector.erl

```diff
@@ -407,6 +407,10 @@ handle_send_result({ok, Reply}) ->
 handle_send_result({error, Reason}) ->
     {error, classify_error(Reason)}.
 
+classify_reply(Reply = #{reason_code := ?RC_PACKET_IDENTIFIER_IN_USE}) ->
+    %% If `emqtt' client restarted, it may re-use packet ids that the remote broker still
+    %% has memory of.  We should retry.
+    {recoverable_error, Reply};
```

Comment:
Hmm, this sounds like a protocol violation. Is it `emqtt` failing proper session recovery under `CleanStart = false`? Probably here, where it relies on `clean_start` flag instead of `session_present`:
```
connected(info, immediate_retry, State = #state{clean_start = false}) ->
```

Tangentially, why does publishing side need `CleanStart = true`? When buffering and retries are involved on connector side, shouldn't this be enough?

Reply:
I think the problem is allowing `clean_start = false` when the emqtt process can be restarted at any point and lose all state such as packet ids.  In this case, I think we ought to retry so we don't silently drop messages in practice.  This came from discussions with @zmstone .

> Tangentially, why does publishing side need CleanStart = true? When buffering and retries are involved on connector side, shouldn't this be enough?

`clean_start = true` is the default, which works much better with the restarting and retrying nature of the buffering layer.  `clean_start = false` + retries is exactly what _could_ lead to a state where the connector is reusing packet ids expected by the remote broker.  If we always start afresh and retry, there's no issue.

Reply:
> `clean_start = false` + retries is exactly what could lead to a state where the connector is reusing packet ids expected by the remote broker.

Right, this matches my understanding, I wonder why it's even possible to configure that (on publishing side).